cura/PrinterOutput/PrintJobOutputModel.py

```diff
@@ -118,17 +118,41 @@ def updateName(self, name: str):
             self.nameChanged.emit()
 
     @pyqtProperty(int, notify = timeTotalChanged)
-    def timeTotal(self):
+    def timeTotal(self) -> int:
         return self._time_total
 
     @pyqtProperty(int, notify = timeElapsedChanged)
-    def timeElapsed(self):
+    def timeElapsed(self) -> int:
         return self._time_elapsed
 
+    @pyqtProperty(int, notify = timeElapsedChanged)
+    def timeRemaining(self) -> int:
+        # Never get a negative time remaining
+        return max(self.timeTotal - self.timeElapsed, 0)
+
+    @pyqtProperty(float, notify = timeElapsedChanged)
+    def progress(self) -> float:
+        result = self.timeElapsed / self.timeTotal
+        if result > 1.0:
+            result = 1.0
+        return result
```

Comment:
It is not a  big deal, but it interesting to see `max()` used above but `min()` isn't being used here.

Reply:
Good point. Don't know why I didn't bother with that. Updated.